### PR TITLE
[Merged by Bors] - Add gossip conditions from spec v0.12.3

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

There are four new conditions introduced in v0.12.3:

 1. _[REJECT]_ The attestation's epoch matches its target -- i.e. `attestation.data.target.epoch ==
  compute_epoch_at_slot(attestation.data.slot)`
1. _[REJECT]_ The attestation's target block is an ancestor of the block named in the LMD vote -- i.e.
  `get_ancestor(store, attestation.data.beacon_block_root, compute_start_slot_at_epoch(attestation.data.target.epoch)) == attestation.data.target.root`
1. _[REJECT]_ The committee index is within the expected range -- i.e. `data.index < get_committee_count_per_slot(state, data.target.epoch)`.
1. _[REJECT]_ The number of aggregation bits matches the committee size -- i.e.
  `len(attestation.aggregation_bits) == len(get_beacon_committee(state, data.slot, data.index))`.

This PR implements new logic to suit (1) and (2). Tests are added for (3) and (4), although they were already implicitly enforced.

## Additional Info

- There's a bit of edge-case with target root verification that I raised here: https://github.com/ethereum/eth2.0-specs/pull/2001#issuecomment-699246659
- I've had to add an `--ignore` to `cargo audit` to get CI to pass. See https://github.com/sigp/lighthouse/issues/1669

### CODE REVIEW

File: beacon_node/beacon_chain/src/attestation_verification.rs [hunk 1]
```diff
@@ -433,16 +459,49 @@ impl<T: BeaconChainTypes> VerifiedUnaggregatedAttestation<T> {
 
         // Check to ensure that the attestation is "unaggregated". I.e., it has exactly one
         // aggregation bit set.
-        let num_aggreagtion_bits = attestation.aggregation_bits.num_set_bits();
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/src/attestation_verification.rs:436]: Glad we finally got this one haha

File: beacon_node/beacon_chain/tests/attestation_verification.rs [hunk 2]
```diff
@@ -654,6 +682,27 @@ fn unaggregated_gossip_verification() {
...
+     *
+     * Spec v0.12.3
+     *
+     * The attestation's epoch matches its target -- i.e. `attestation.data.target.epoch ==
+     *   compute_epoch_at_slot(attestation.data.slot)`
+     *
+     */
+
+    assert_invalid!(
+        "attestation without invalid target epoch",
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/tests/attestation_verification.rs:696]:
```suggestion
        "attestation with invalid target epoch",
```

**Review by michaelsproul — APPROVED** [review]
Looks good to me.

Just one tiny nitpick.

I wonder if we should hold off merging temporarily until we reach a resolution on the attestation-to-future-blocks issue?

**Comment by paulhauner** [comment]
> I wonder if we should hold off merging temporarily until we reach a resolution on the attestation-to-future-blocks issue?

I feel it's fine to merge without a resolution. The head-block check wasn't included in v0.12.2, so we're not really worse off. I could be persuaded otherwise, though.

**Comment by michaelsproul** [comment]
Yeah I'm happy with that, feel free to merge
